astropy/nddata/utils.py

```diff
@@ -242,8 +244,14 @@ def extract_array(array_large, shape, position, mode='partial',
 
     # Extracting on the edges is presumably a rare case, so treat special here
     if (extracted_array.shape != shape) and (mode == 'partial'):
-        extracted_array = np.zeros(shape, dtype=array_large.dtype)
-        extracted_array[:] = fill_value
+        try:
+            extracted_array = np.zeros(shape, dtype=array_large.dtype)
```

Comment:
Maybe move the definition out of the `try/except`.

Reply:
Done